pandas/tests/indexes/test_range.py

```diff
@@ -241,6 +241,36 @@ def test_view(self):
     def test_dtype(self):
         assert self.index.dtype == np.int64
 
+    def test_has_called_data(self):
+        # Calling RangeIndex._data caches a array of the same length.
+        # This tests whether RangeIndex._data has been called by doing methods
+        idx = RangeIndex(0, 100, 10)
+        assert idx._has_called_data is False
```

Comment:
I would suggest that you monkeypatch the class here, a bit cleaner as the code then doesn't have this attribute

Reply:
The attribute is a property (previously ``cache_readonly``) and in neither cases is it technically possible to dynamically monkey-patch ``_data``. I could subclass ``RangeIndex`` and add a new property, but not sure if that's better than this?